src/MiddlewareListener.php

```diff
@@ -91,6 +90,8 @@ public function onDispatch(MvcEvent $event)
             }
         }
 
+        $event->setError('');
```

Comment:
I'd argue both the `DispatchListener` and `MiddlewareListener` should act that way, with the following at the top of the listeners:

```php
if (null !== $e->getResult()) {
    return;
}
```

Do you want to incorporate that change in this PR?